### DISCUSSION TITLE
"must be a string" error when running docker compose

### DISCUSSION THREAD

**Discussion by riemino** [apache/superset#28184] [General]
It seems like the Quick Start documentation has changed recently. I was following the instructions (https://superset.apache.org/docs/quickstart) on setting up Superset on Ubuntu VM.  After I clone the repo, I run the docker compose command: 

> docker compose -f docker-compose-image-tag.yml up 

and get the following error: 

> validating /home/administrator/appsroot/Superset/superset/docker-compose-image-tag.yml: services.db.env_file.0 must be a string

What am I missing? 

Thank you!

**Comment by guldnerthomas** [comment]
I don't know the issue, but you can replace in docker-compose-image-tag.yml :
```
    env_file:
      # defaults
      - docker/.env
      # gitignored overrides
      - docker/.env-local
```
 by :
```
env_file: docker/.env
```

**Reply by riemino** [reply]
Thanks very much 👍🏼

**Reply by pawelofficial** [reply]
sorry for a silly question - im facing the same issue and i cant fix it 

ok i think i made it work, the thing was to edit the docker-compose-image-tag.yml
and change syntax from 

env_file:
    path: docker/.env-local

into 

    env_file:
      - docker/.env

**Comment by ogirardot** [comment]
upgrade to the latest docker version, docker compose syntax evolved

**Reply by riemino** [reply]
👍🏼

**Reply by marttini** [reply]
it worked for me

**Comment by dziamid** [comment]
Got cought by this as well. According to [docker compose docs](https://docs.docker.com/compose/compose-file/05-services/)  `required` attribute is available in Docker Compose version 2.24.0 or later.
